pkg/restore/admissionwebhook_config_action.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restore
+
+import (
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+)
+
+// AdmissionWebhookConfigurationAction is a RestoreItemAction plugin applicable to mutatingwebhookconfiguration and
+// validatingwebhookconfiguration to reset the invalid value for "sideEffects" of the webhooks.
+// More background please refer to https://github.com/vmware-tanzu/velero/issues/3516
+type AdmissionWebhookConfigurationAction struct {
+	logger logrus.FieldLogger
+}
+
+// NewAdmissionWebhookConfigurationAction creates a new instance of AdmissionWebhookConfigurationAction
+func NewAdmissionWebhookConfigurationAction(logger logrus.FieldLogger) *AdmissionWebhookConfigurationAction {
+	return &AdmissionWebhookConfigurationAction{logger: logger}
+}
+
+// AppliesTo implements the RestoreItemAction plugin interface method.
+func (a *AdmissionWebhookConfigurationAction) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"mutatingwebhookconfigurations", "validatingwebhookconfigurations"},
+	}, nil
+}
+
+// Execute will reset the value of "sideEffects" attribute of each item in the "webhooks" list to "None" if they are invalid values for
+// v1, such as "Unknown" or "Some"
+func (a *AdmissionWebhookConfigurationAction) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
+	a.logger.Info("Executing ChangeStorageClassAction")
+	defer a.logger.Info("Done executing ChangeStorageClassAction")
+
+	item := input.Item
+	apiVersion, _, err := unstructured.NestedString(item.UnstructuredContent(), "apiVersion")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get the apiVersion from input item")
+	}
+	name, _, _ := unstructured.NestedString(item.UnstructuredContent(), "metadata", "name")
+	logger := a.logger.WithField("resource_name", name)
+	if apiVersion != "admissionregistration.k8s.io/v1" {
```

Comment:
Do we have to take care `admissionregistration.k8s.io/v1beta1` which is < 1.16?
Or no, Velero only support 1.16+ in the next coming release?

Reply:
velero v1.8+ will only work on k8s 1.16+

But this check 
```
if apiVersion != "admissionregistration.k8s.io/v1"  ...
```
means this plugin will only handle the `v1` resources with illegal values in `sideEffects` attributes.  So when it sees v1beta1 resource it just skips it.